### ISSUE TITLE
parser.parse is not a function

### ISSUE THREAD

**Issue by caribou-code** [stylelint/stylelint#4182]
I have a recurring issue with my file `components/Label/index.style.ts` below:

```
import styled from 'styled-components';

export const Thing = styled.div`
  color: ${p => p.labelTextColor};
`;
```

Running `stylelint 'components/Label/index.style.ts'` on this file throws the following error:

```
TypeError: parser.parse is not a function
    at Object.templateParse [as parse] (/Users/.../Website/node_modules/postcss-jsx/template-parse.js:10:9)
...
```

My project uses Styled Components, React and Typescript and the `.stylelintrc` looks like this:

```
{
  "extends": [
    "stylelint-config-standard",
    "stylelint-config-styled-components",
    "stylelint-config-rational-order"
  ],
  "rules": {
    "declaration-empty-line-before": null
  }
}

```
 
I can consistently replicate the issue with this file, but other files that looks almost identical (using props and arrow functions etc) do not throw this stylelint error, so I'm really not sure what's going on here since I'm unable to replicate the issue in other files.

Package versions I'm using are:

```
"stylelint": "10.1.0",
"stylelint-config-rational-order": "0.1.2",
"stylelint-config-standard": "18.3.0",
"stylelint-config-styled-components": "0.1.1",
"stylelint-order": "3.0.1",
```

**Comment by hudochenkov** [comment]
Could you create a repository with this problem, please?

**Comment by OTL-WOOSOL** [comment]
How did you solve this problem? I'm going crazy.

**Comment by JounQin** [comment]
> How did you solve this problem? I'm going crazy.

https://antfu.me/posts/why-reproductions-are-required

**Comment by FANZHETWO** [comment]
You should check if the postcss version is 6.0.0 or higher
